grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -89,6 +94,10 @@ class LocationDataService {
             if (organizations?.size() > 1) {
                 command.errors.reject("Row ${index + 1}: '${organizations.size()}' records found for organization name '${params.organization}'. Please specify by entering the organization code instead")
             }
+
+            if (params.locationType && locationToBeUpdated && (locationToBeUpdated.locationType.isInternalLocation() || locationToBeUpdated.locationType.isZone()) && !(LocationType.findByNameLike(params.locationType + "%").isInternalLocation() || LocationType.findByNameLike(params.locationType + "%").isZone())) {
```

Comment:
It is because here I'm checking if type of existing location isn't changed from internal to non-internal location type, so I have to check if existing location is internal and then check if new location type is also internal.

Reply:
Yeah, but instead of this `LocationType.findByNameLike(params.locationType + "%")` can't you use the `locationType` already defined? And why there is looking by name with wildcard `%`?

Reply:
It was already done before, maybe it is because values are like "Supplier|fr:Fournisseurs" or "Pharmacy|fr:Pharmacy".

Reply:
@MagdaToczek Makes sense, but then should not you look for existing location type with the same way?

Reply:
Ok, I'll change it in existing location.